pkg/clients/topic/topic.go

```diff
@@ -0,0 +1,334 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topic
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/reference"
+
+	arv1 "github.com/IBM/eventstreams-go-sdk/pkg/adminrestv1"
+
+	"github.com/crossplane-contrib/provider-ibm-cloud/apis/eventstreamsadminv1/v1alpha1"
+	ibmc "github.com/crossplane-contrib/provider-ibm-cloud/pkg/clients"
+)
+
+// LateInitializeSpec fills optional and unassigned fields with the values in *arv1.TopicDetail object.
+func LateInitializeSpec(spec *v1alpha1.TopicParameters, in *arv1.TopicDetail) error { // nolint:gocyclo
+
+	if spec.PartitionCount != nil {
+		spec.Partitions = spec.PartitionCount
+	}
+
+	if spec.Partitions == nil {
+		spec.Partitions = in.Partitions
+	}
+
+	if spec.PartitionCount == nil {
+		spec.PartitionCount = in.Partitions
+	}
+
+	// this works but it still updates configs everytime ?? could potentially optimize but would be difficult and probably increase cyclomatic complexity ??
```

Comment:
yeah, we are somehow constrained by the way the event streams topic API is designed/implemented. I think it trying to further optimize would yield diminishing return, so I would leave optimizations for later if we find issues.